### PR TITLE
Add bernoulli to ATen XLA tensor

### PR DESCRIPTION
**Pull request by asuhan**

### CODE REVIEW

File: torch_xla/csrc/tensor_methods.cpp [hunk 1]
```diff
@@ -514,6 +514,28 @@ XLATensor XLATensor::batch_norm(const XLATensor& input, const XLATensor& weight,
       input.GetIrValue(), weight_value, bias_value, momentum, eps));
 }
 
+XLATensor XLATensor::bernoulli(const XLATensor& input, double probability) {
+  return input.CreateFrom(ir::ops::Bernoulli(
+      input.GetIrValue(),
+      ir::MakeNode<ir::ops::Scalar>(probability, input.shape())));
```
Comment (dlibenzi) [inline torch_xla/csrc/tensor_methods.cpp]: GetIrValueForScalar().
Same blow.

File: torch_xla/csrc/xla_lower_util.cpp [hunk 2]
```diff
@@ -243,17 +243,23 @@ xla::XlaOp CreateMatMul(const xla::XlaOp& lhs, const xla::XlaOp& rhs) {
               << rhs_shape << ")";
 }
 
+xla::XlaOp BuildBernoulli(const xla::XlaOp& probability,
+                          const xla::Shape& shape) {
+  xla::Shape probability_shape = XlaHelpers::ShapeOfXlaOp(probability);
```
Comment (dlibenzi) [inline torch_xla/csrc/xla_lower_util.cpp]: probablity_shape is always shape.

Comment (asuhan) [reply torch_xla/csrc/xla_lower_util.cpp]: What is your suggestion here?

Comment (asuhan) [reply torch_xla/csrc/xla_lower_util.cpp]: If your claim is that `shape` and `probability_shape` are always the same, I don't think they are. They can differ in element type: for example, `XLATensor::bernoulli_` preserves the shape of input and takes the probability from a different tensor. They are only guaranteed to be the same for `XLATensor::bernoulli`.

Comment (dlibenzi) [reply torch_xla/csrc/xla_lower_util.cpp]: I am not in love with the fact that in many places we are coming here, we know that probability_shape is shape, and we issue a call to XlaHelpers::ShapeOfXlaOp to get it.
Give that we are generating a noise from 0 to 1, I think we can pun to F32 as noise type (which is only used to build the mask).
At that point we need no probability_shape anymore as all the paths leading here have same dimensions.

Comment (asuhan) [reply torch_xla/csrc/xla_lower_util.cpp]: Making the noise F32 sounds ok, I'll change it.

File: torch_xla/csrc/aten_xla_type.cpp [hunk 3]
```diff
@@ -648,6 +648,44 @@ at::Tensor AtenXlaType::batch_norm(
       bridge::GetOrCreateXlaTensor(bias, device), momentum, eps));
 }
 
+at::Tensor AtenXlaType::bernoulli(const at::Tensor& self, double p,
+                                  at::Generator* generator) const {
+  if (generator) {
```
[the remark this answers is not in the document]
Comment (asuhan) [reply torch_xla/csrc/aten_xla_type.cpp]: Done.

File: torch_xla/csrc/aten_xla_type.cpp [hunk 4]
```diff
@@ -648,6 +648,44 @@ at::Tensor AtenXlaType::batch_norm(
...
+    return AtenXlaTypeBase::bernoulli(self, p, generator);
+  }
+  return bridge::AtenFromXlaTensor(
+      XLATensor::bernoulli(bridge::GetXlaTensor(self), p));
+}
+
+at::Tensor AtenXlaType::bernoulli(const at::Tensor& self,
+                                  at::Generator* generator) const {
+  if (generator) {
```
[the remark this answers is not in the document]
Comment (asuhan) [reply torch_xla/csrc/aten_xla_type.cpp]: Done.

File: torch_xla/csrc/aten_xla_type.cpp [hunk 5]
```diff
@@ -648,6 +648,44 @@ at::Tensor AtenXlaType::batch_norm(
...
+    return AtenXlaTypeBase::bernoulli(self, generator);
+  }
+  XLATensor self_tensor = bridge::GetXlaTensor(self);
+  return bridge::AtenFromXlaTensor(XLATensor::bernoulli(self_tensor));
+}
+
+at::Tensor& AtenXlaType::bernoulli_(at::Tensor& self, double p,
+                                    at::Generator* generator) const {
+  if (generator) {
```
[the remark this answers is not in the document]
Comment (asuhan) [reply torch_xla/csrc/aten_xla_type.cpp]: Done.

File: torch_xla/csrc/aten_xla_type.cpp [hunk 6]
```diff
@@ -648,6 +648,44 @@ at::Tensor AtenXlaType::batch_norm(
...
+    return AtenXlaTypeBase::bernoulli_(self, p, generator);
+  }
+  XLATensor self_tensor = bridge::GetXlaTensor(self);
+  XLATensor::bernoulli_(self_tensor, p);
+  return self;
+}
+
+at::Tensor& AtenXlaType::bernoulli_(at::Tensor& self, const at::Tensor& p,
+                                    at::Generator* generator) const {
+  if (generator) {
```
[the remark this answers is not in the document]
Comment (asuhan) [reply torch_xla/csrc/aten_xla_type.cpp]: Done.

File: torch_xla/csrc/tensor_methods.cpp (hunk 1 above)
Comment (asuhan) [reply torch_xla/csrc/tensor_methods.cpp]: Done.

File: torch_xla/csrc/xla_lower_util.cpp (hunk 2 above)
Comment (asuhan) [reply torch_xla/csrc/xla_lower_util.cpp]: Not sure what you meant here. We need a shape for `xla::RngUniform` and `shape` can have any element type. Would you rather create `probability_shape` from `shape`, with `F32` as an element type? That'd avoid querying it, but I don't understand why querying it is bad.

Comment (dlibenzi) [reply torch_xla/csrc/xla_lower_util.cpp]: OK, leave it as is.

**Pull request merged**